### PR TITLE
Install the gem directly in the providers

### PR DESCRIPTION
**Pull request by tas50**
This should probably get pulled into the AWS module and included from there, but for now it's a huge win in that we no longer have to include the default recipe just to install the gem.  The default recipe is still there if you want to ensure that upgrades occur.  If you're not concern about that aka you don't have legacy installs then don't use it.

### CODE REVIEW

File: libraries/aws.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+module Opscode
+  module Aws
+    def require_aws_sdk
+      # require the version of the aws-sdk specified in the node attribute
+      require 'rubygems'
```
Comment (lamont-granquist) [inline libraries/aws.rb]: this hasn't been necessary since we dropped 1.8.7 support

**Comment by lamont-granquist** [comment]
:+1: 

only you didn't go far enough, we should bake the most recent aws-sdk gem into the omnibus-chef install...

**Comment by lamont-granquist** [comment]
also :+1: on use module code to do this and not attempting to `include_recipe` the default recipe from a provider, that's always kind of a shit-show...

**Comment by tas50** [comment]
Thought about the recipe method, but I wanted the recipe to go away and I figured we already had a pile of libraries.  Might as well add one more.

**Comment by lamont-granquist** [comment]
i think you just invented a new best practice...

**Pull request merged**